### PR TITLE
Add support for comparison with null parameters

### LINKED ISSUE #153: Add support for comparison of values to null
Hi, I'd like to discuss implementing comparison of values to null. In this fork https://github.com/sklose/NCalc2/ there is `EvaluateOptions.AllowNullParameter` option, we're actively using it. Would like to port this feature to NCalc.

### PR DESCRIPTION
**Pull request by Bykiev**
This PR adds support for comparision with null parameters. To compare with null you should use ExpressionOptions.AllowNullParameter

Closes #153

### CODE REVIEW

File: src/NCalc/ExpressionOptions.cs
Comment (gumbarros) [inline src/NCalc/ExpressionOptions.cs:1]: Nice

**Comment by gumbarros** [comment]
Instead of merging on `main`, let's create a v4.1?

**Comment by Bykiev** [comment]
> Instead of merging on `main`, let's create a v4.1?

Sounds good, but I'm not good with a git, can you please help me with it?

**Comment by gumbarros** [comment]
Also, can you add to docs this feature at parameters.md please?

**Comment by gumbarros** [comment]
> > Instead of merging on `main`, let's create a v4.1?
> 
> Sounds good, but I'm not good with a git, can you please help me with it?

Yeah, it's on my side before merging, no problem

**Comment by gumbarros** [comment]
image
Not very familiar with nbgv (the tool used on this repo), but I think we are good now, master is the preview branch.

**Comment by Bykiev** [comment]
> Also, can you add to docs this feature at parameters.md please?

Done, please check the grammar, I'm not good in English ;)

**Review by gumbarros — APPROVED** [review]

**Comment by gumbarros** [comment]
> The build is failed for some reason...

At your machine or CI? GitHub actions seems fine, just waiting AppVeyor before merging

**Pull request merged**